src/proto/hybrid_vec.rs

```diff
@@ -0,0 +1,1832 @@
+use alloc::heap::{EMPTY, allocate, reallocate, deallocate};
+
+// Implemented on top of `core` to demonstrate the ability to substitute HybridVec for Vec
+use core::borrow::{BorrowFrom, ToOwned};
+use core::default::Default;
+use core::fmt;
```

Comment:
Just a nit, but I always like to group imports like this into `use core::{fmt, inter, mem, ops, ptr, usize};` instead of a bunch of different `use`s.